### DISCUSSION TITLE
Auto-sync shared dictionary

### DISCUSSION THREAD

**Discussion by user202729** [opensteno/plover#1383] [Ideas]
I'm thinking of a dictionary plugin such that if someone add a new entry to their dictionary, it would be synchronized with the other people using the same dictionary.

The disadvantage would be that some people have their modification of the theory, and it would be inconsistent with the rest. But other than the occasional word boundary conflict, if someone want to modify the dictionary such that conflicts with another modification, it would be stored in their "local dictionary".

Is anyone interested in the idea?

**Comment by mawillcockson** [comment]
I'm very interested in this idea, but more for keeping the `main.json` dictionary up to date separate from updating Plover itself.

**Reply by user202729** [reply]
The problem of main.json at the moment is that the developers have been leaving #400 unaddressed for a while.

This idea is only useful if there's a group of people using Plover theory and use the plugin at the same time.

Depends on the need, a shared Google Drive/something-drive might be sufficient.

**Reply by mawillcockson** [reply]
They still do occasionally make changes.

In general, it'd be nice if there was dictionary management equivalent to current plugin management:

- download new dictionary and any required plugins
- check upstream source for changes, and notify when updates are available

If people can make changes to dictionaries downloaded in this way, then updating without overwriting user's changes would be tough.

If the general guidance of "put dictionary overrides in `user.json`" is followed, this is easier, but a per-dictionary override system might be nice.

In general, though, just the first two points would be awesome, if changes to the dictionary can be prevented.
